src/vs/workbench/api/common/extHostTreeViews.ts

```diff
@@ -237,7 +239,7 @@ class ExtHostTreeView<T> extends Disposable {
 				result.message = true;
 			}
 			return result;
-		}, 200)(({ message, elements }) => {
+		}, 200, this.leadingDebounce)(({ message, elements }) => {
```

Comment:
Why not just make this true?

Reply:
@alexr00  I didn't make it true to simplify unit tests. I can make it true, then the change will grow to update all of them, and make them more complicated. I think, having the flag for unit tests is justifiable here.

Reply:
@alexr00 If you think, it makes sense to change tests for this, I can make the change, but overall, I think, it's better to keep this as a setting.

Reply:
@alexr00 So what do you think? Does it makes sense?

Reply:
With this flag, we are not testing the real behavior.
I think that allowing the leading event could have a negative impact for extensions that use the tree `message`. If they trigger a treedata change and also a message change, then we do still want to debounce the first event to keep the message change happening as close to the data change as possible. Having these be out of sync will be a more noticeable UX problem than delaying first event by 200ms.

I am open to other arguments, so please feel free to make them!

Reply:
@alexr00 It's a tradeoff. You make life of extension authors better in cases when they fire many events, but make the life of extension authors who need responsive UI, and are ready to optimize event firing worse.

What do you think about adding an option to tree registration method to add/remove leaedingDebounce? I can add it to just unstable API for now.

Reply:
I don't think we should have API for adding/removing the leading debounce, this isn't something extension authors should need to think about.
I tested out how `leading` impacts the `message` setting in the References view, and it actually improves things. This will be view implementation dependent, but we can try it out and see if it causes problems for anyone. 
Please just update the tests and set `leading` to true and I'll review again!